internal/cli/root/builder.go

```diff
@@ -96,3 +116,56 @@ func formattedVersion() string {
 		runtime.GOARCH,
 		runtime.Compiler)
 }
+
+func (opts *BuilderOpts) hasNewVersionAvailable() (newVersionAvailable bool, newVersion string, err error) {
+	if version.Version == "" {
+		return false, "", nil
+	}
+
+	svCurrentVersion, err := semver.NewVersion(version.Version)
+	if err != nil {
+		return false, "", err
+	}
+
+	latestVersion, err := opts.store.LatestVersion()
+	if err != nil {
+		return false, "", err
+	}
+
+	svLatestVersion, err := semver.NewVersion(latestVersion)
+	if err != nil {
+		return false, "", err
+	}
+
+	if svCurrentVersion.Compare(svLatestVersion) < 0 {
+		return true, latestVersion, nil
+	}
+
+	return false, "", nil
+}
+
+func shouldSkipPrintNewVersion(w io.Writer) bool {
+	return config.SkipUpdateCheck() || cli.IsTerminal(w)
+}
+
+func (opts *BuilderOpts) printNewVersionAvailable(w io.Writer) error {
+	newVersionAvailable, latestVersion, err := opts.hasNewVersionAvailable()
+	if err != nil {
+		return err
+	}
+	if newVersionAvailable {
+		newVersionTemplate := `
+A new MongoDB CLI version is available '%s'!
+To upgrade MongoCLI, see https://dochub.mongodb.org/core/mongocli-install. To disable this alert, run "mongocli config set skip_update_check true".
```

Comment:
We're not naming the product consistently here -- it's probably better to only have the name once.  

How about: 
```
A new version of mongocli is available! 
To upgrade, see: 

To disable this alert (...) 
```

nit: some vertical space between "to upgrade" and "to disable" would be good